pkg/docker/config/config.go

```diff
@@ -305,6 +306,11 @@ func findAuthentication(registry, path string, legacyFormat bool) (string, strin
 		return getAuthFromCredHelper(ch, registry)
 	}
 
+	// Second try default credential store
+	if cs := auths.CredStore; cs != "" {
+		return getAuthFromCredHelper(cs, registry)
+	}
+
```

Comment:
At least `SetAuthentication` and `RemoveAuthentication`, and maybe also `RemoveAllAuthentication` (not at all sure now) needs to be updated as well.

Reply:
Updated, fixed the order of importance as well.